src/utils/listenerHelpers.ts

```diff
@@ -0,0 +1,97 @@
+import { DispatchedEvent } from "globals/Calc";
+import { Calc } from "globals/window";
+
+const dispatchOverridingHandlers: [
+  (evt: DispatchedEvent) => boolean | undefined,
+  number,
+  number
+][] = [];
+
+let dispatchOverridingHandlerId = 0;
+
+// schedule a function to run after every desmos event
+// priorities determine which run first
+// the handler can return false to force the dispatcher to stop early
+// (e.g. to stop desmos from doing a default action upon pressing a key)
+export function registerCustomDispatchOverridingHandler(
+  handler: (evt: DispatchedEvent) => boolean | undefined,
+  priority: number
+): number {
+  const id = dispatchOverridingHandlerId++;
+  // add the handler
+  dispatchOverridingHandlers.push([handler, priority, id]);
+
+  // sort the handlers so that higher priorities are first
+  // could easily be optimized but prob not a bottleneck
+  dispatchOverridingHandlers.sort((a, b) => b[1] - a[1]);
+
+  return id;
+}
+
+// deregisters a function created with registerCustomDispatchOverridingHandler
+// uses the id that the former function returns
+export function deregisterCustomDispatchOverridingHandler(id: number): void {
+  // find the handler by id
+  const firstIndex = dispatchOverridingHandlers.findIndex(
+    ([_, __, id2]) => id2 === id
+  );
+  if (firstIndex === -1) {
+    // eslint-disable-next-line no-console
+    console.error(
+      `Failed to deregister the dispatch overriding handler '${id}'.`
+    );
+  }
+
+  // remove the handler
+  dispatchOverridingHandlers.splice(firstIndex, 1);
+}
+
+// once Calc is defined, change handleDispatchedAction to first
+// run a set of custom handlers
+const interval = setInterval(() => {
+  if (!Calc) return;
+  clearInterval(interval);
```

Comment:
`pollForValue` from `utils.ts` would save some of this code, but it would be better to either:
- wrap this in a new function `setupDispatchOverride()` and call that from `runDesModder()` in preload/script
- change this to be a core plugin, similar to manage-metadata or pillbox-menus. Then you get guaranteed that it runs after `Calc` is defined, and you have a clear-cut way to keep track of the state (`dispatchOverridingHandlers` etc)

Btw this is what I referred to with the top-level stuff. The `handleDispatchedAction` modification is top-level, so you were surprised by when it ran (before `Calc` was defined) and had to do some hacks to get it working). The two ideas above involve wrapping this setup stuff in a function or class so it's easier to control when it runs.

Reply:
Fixed by calling it in preload.